runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -458,11 +450,14 @@ class MM_ObjectAccessBarrierAPI
 					offset += sizeof(uintptr_t);
 				}
 			}
-		
-			/* zero lockword, if present */
-			j9objectmonitor_t *lockwordAddress = getLockwordAddress(vmThread, destObject);
-			if (NULL != lockwordAddress) {
-				J9_STORE_LOCKWORD(vmThread, lockwordAddress, 0);
+
+			if (initializeLockWord) {
```

Comment:
again, is this just unrelated optimization (based on premise that most of time objects do not have a lock), or really a required change for heapify?